### PR TITLE
GeneralWallet supports delete_wallet and reset_wallet_info

### PR DESCRIPTION
**Pull request by huazhouwang**
## What does this implement/fix? Explain your changes.
1. delete_wallet 需要判断 hd 然后清空所有主钱包
2. reset_wallet_info 支持清除 App 数据

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
…

## Pull request type
 
_Put an `x` in the boxes that apply_
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
…

### CODE REVIEW

File: electrum_gui/common/the_begging.py [hunk 1]
```diff
@@ -29,3 +29,14 @@ def terminate():
...
+
+
+def reset_runtime():
+    import os
+
+    from electrum_gui.common.basic.orm import database
+
+    terminate()
+    os.remove(database.db.database)
+    database.db.close()
```
Comment (taimanhui) [inline electrum_gui/common/the_begging.py:41]: 这两步是不是要反过来？或者是文件名先取出来然后关掉数据库再删文件。

吐槽下这个文件名的变量属性名居然是`database`这么迷惑...

File: electrum_gui/common/wallet/manager.py [hunk 2]
```diff
@@ -797,11 +797,29 @@ def cascade_delete_wallet_related_models(wallet_id: int, password: str = None):
...
         check_wallet_password(wallet_id, password)
 
+    chain_code = wallet.chain_code
     accounts = daos.account.query_accounts_by_wallets([wallet_id])
+    addresses = [i.address for i in accounts]
     related_pubkey_ids = {i.pubkey_id for i in accounts if i.pubkey_id is not None}
     if related_pubkey_ids:
         secret_manager.cascade_delete_related_models_by_pubkey_ids(list(related_pubkey_ids))
 
+    transaction_manager.delete_actions_by_addresses(chain_code, addresses)
```
Comment (taimanhui) [inline electrum_gui/common/wallet/manager.py:807]: 这个应该是之前都缺的这里补上，和这个清理hd wallet相关性不高对吧？

File: electrum_gui/common/wallet/manager.py [hunk 3]
```diff
@@ -797,11 +797,29 @@ def cascade_delete_wallet_related_models(wallet_id: int, password: str = None):
...
+@_require_primary_wallet_exists()
+def get_first_primary_wallet_id() -> int:
+    return daos.wallet.get_first_primary_wallet().id
+
+
+@db.atomic()
+def clear_all_primary_wallets(password: str = None):
+    wallets = daos.wallet.list_all_wallets(wallet_type=WalletType.SOFTWARE_PRIMARY)
+    if not wallets:
+        return
```
Comment (taimanhui) [inline electrum_gui/common/wallet/manager.py:822]: 如果是空列表的话可以不用单独判断，下面的代码一样能正常运行

**Review by taimanhui — DISMISSED** [review]

File: electrum_gui/common/wallet/manager.py (hunk 2 above)
Comment (huazhouwang) [reply electrum_gui/common/wallet/manager.py:807]: 嗯，之前忘记加了，补上

File: electrum_gui/common/the_begging.py [hunk 4]
```diff
@@ -29,3 +29,14 @@ def terminate():
...
+
+
+def reset_runtime():
+    import os
+
+    from electrum_gui.common.basic.orm import database
+
+    terminate()
+    database.db.close()
+    os.remove(database.db.database)
```
Comment (taimanhui) [inline electrum_gui/common/the_begging.py:41]: 先关闭不影响文件名获取对吧，那就没问题

**Review by taimanhui — APPROVED** [review]

File: electrum_gui/common/the_begging.py (hunk 4 above)
Comment (huazhouwang) [reply electrum_gui/common/the_begging.py:41]: 嗯，不影响，测试过了

**Review by shuaiLiWang — APPROVED** [review]

**Pull request merged**